### PR TITLE
Fix Lodash isArray deprecation

### PR DESCRIPTION
**Pull request by damassi**
When installing the latest danger, I noticed this in the console output: 

screen shot 2017-03-29 at 5 15 32 pm

Replaces `lodash.isarray` with native implementation.

### CODE REVIEW

File: source/platforms/github/GitHubGit.ts [hunk 1]
```diff
@@ -114,7 +113,7 @@ export default async function gitDSLForGitHub(api: GitHubAPI): Promise<GitDSL> {
       // added or removed. This makes it really easy to act on specific
       // changes to JSON DSLs
 
-      if (isarray(diff.after) && isarray(diff.before)) {
+      if (Array.isArray(diff.after) && Array.isArray(diff.before)) {
```
Comment (macklinu) [inline source/platforms/github/GitHubGit.ts]: I think we should be safe using the native `Array.isArray()` instead of that deprecated lodash package. 👍

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "danger",
-  "version": "0.14.1",
+  "version": "0.14.2",
```
Comment (macklinu) [inline package.json]: Could you revert this change please? Unless we are good to ship this in a patch release - cc @orta

File: changelog.md [hunk 3]
```diff
@@ -4,10 +4,11 @@
 
 ### 0.14.2
 
-* Moved `@types/chalk` from dependencies to devDependencies - orta
+* Replaced deprecated `lodash.isarray` package with `Array.isArray` - damassi
 
 ### 0.14.1
 
+* Moved `@types/chalk` from dependencies to devDependencies - orta
```
Comment (macklinu) [inline changelog.md]: Good call, I think this was shipped in 0.14.1.

File: changelog.md [hunk 4]
```diff
@@ -4,10 +4,11 @@
 
 ### 0.14.2
 
-* Moved `@types/chalk` from dependencies to devDependencies - orta
+* Replaced deprecated `lodash.isarray` package with `Array.isArray` - damassi
```
Comment (macklinu) [inline changelog.md]: Looking for confirmation from @orta on shipping 0.14.2 or holding off on it for anything else to keep this changelog message here. Otherwise we usually add changelog messages right below the `//  Add your own contribution below` line and then add the version number we are shipping to the changelog as we package up a release.

**Comment by macklinu** [comment]
Forgot to add, Danger commented on the PR requesting that `yarn install` is run (for dependency additions/removals to keep `yarn.lock` in sync with `package.json`). Could you please run that locally and push up a commit with the changes to `yarn.lock`? 🙂

File: changelog.md (hunk 4 above)
Comment (damassi) [inline changelog.md]: Ah, I see -- `danger` said to update the changelog and so I followed the format as it was there. It might be good to provide a little more direction in terms of version numbers, changelog, etc.

**Comment by damassi** [comment]
Updated and rebased. It's odd -- I ran a `yarn install` first thing after pulling down the package, but I had to `rm` it and reinstall packages before it would update.

**Comment by orta** [comment]
Ah yeah, the version/changelog thing - again sorry about that, :D

I got stuck on a particularly long PR, which I thought I could get finished by the weekend ( https://github.com/danger/danger-js/pull/194 ) but inevitably I couldn't.

IMO it makes sense to be more verbose in the changelog about saying above the latest released version. 👍

**Comment by orta** [comment]
Shipping `0.14.2` sounds 👍

**Comment by orta** [comment]
@damassi with respect to the lockfile,  you would have needed to run `yarn remove lodash-isarray` this would have removed the module then only re-created the dependency locks for anything related to it.

In deleting and re-creating it from scratch you've updated hundreds of dependencies in the stack. 

I'm ok with this, after eyeing it over, but ideally next time we should be a bit more strict about smaller lock file changes.

**Comment by orta** [comment]
Alright, I've shipped that - and all other PRs as 0.14.2 

I've also updated the notice on the CHANGELOG to be more specific in https://github.com/danger/danger-js/commit/fde4c54d4e01d4f34921aec43624780b2410458e

**Comment by damassi** [comment]
Ah sorry about that! I wasn't aware of `yarn remove <package>`. Noted 👍
